test/specs/reference/spec.js

```diff
@@ -0,0 +1,194 @@
+var includes = require('./../../utils/ermrest-init.js').init();
+
+var server = includes.server;
+var ermRest = includes.ermRest;
+var ermrestUtils = includes.ermrestUtils;
+
+describe("For determining reference objects and it's child objects,", function () {
```

Comment:
As mentioned earlier for `parser` spec, you need to follow the folder structure as mentioned in the [wiki](https://github.com/informatics-isi-edu/ermrestjs/wiki/Unit-Testing-In-ermrestjs).

Apart from that, you can always access the **catalogId** using **process.env.DEFAULT_CATALOG** anywhere in your test file.

For other variables you can hardcode the values. To access the `server`, `ermRest` and other variables you can have a `beforeAll` block, which reads the values from the `options` variable passed to the **execute** function in the test file.

For more info you can refer this [link](https://github.com/informatics-isi-edu/ermrestjs/wiki/Unit-Testing-In-ermrestjs#2-test-spec)